app/Services/PirepService.php

```diff
@@ -489,6 +489,8 @@ public function submit(Pirep $pirep)
         // only update the pilot last state if they are accepted
         if ($default_state === PirepState::ACCEPTED) {
             $pirep = $this->accept($pirep);
+        } elseif ($default_state === PirepState::REJECTED) {
+			event(new PirepRejected($pirep));
```

Comment:
Why not just call `$this->reject($pirep);` ? This will send the event, but it will also perform the other actions needed when rejecting a PIREP. This is the function that is called when a PIREP is rejected from the admin panel.

Reply:
I explained why, I don't know say you more because my technical knowledge is not as good as yours.

The autoreject pireps via listeners isn't sending the reject email, because core only have implemented **auto accept**, **pending** (default state) or **manual reject** via admin panel (only rejecting in this way a reject email is sent to pilot), but **not work** with listener in "pirep filed" changing state to reject (to autoreject pireps).

With this solution that worked, changing state in listener and sending a reject email.

This was the chat about this: [https://discord.com/channels/332920618729340929/780945843654098944/1263458136505847942](https://discord.com/channels/332920618729340929/780945843654098944/1263458136505847942)

Reply:
> Why not just `$pirep = $this->reject($pirep);` ? This will send the event, but it will also perform the other actions needed when rejecting a PIREP. This is the function that is called when a PIREP is rejected from the admin panel.

Simple; Because we do not need it there, what needed is the notification and that functions does a lot more.

Reply:
@arthurpar06 Sorry, I just copy pasted it from discord 😉 

![image](https://github.com/user-attachments/assets/7cd4f5d9-ce05-4151-8390-73bcc1f3bf53)